### DISCUSSION TITLE
When finding and executing an existing queryFn with queryKeyHash or queryKeyHashFn, please modify the queryFn parameter so that the queryKey contains the queryKey of the useQuery that executed the queryKeyHash/queryKeyHashFn.

### DISCUSSION THREAD

**Discussion by Wick-Lim** [TanStack/query#7698] [General]
I would like to create a useQuery that corresponds to all APIs in one place, and use only the queryKey on each page. However, in cases where values ​​such as searchParams need to be passed as queryKey, the queryFn was made to be executed through queryKeyHash or queryKeyHashFn. However, since the original queryKey is not known, the parameter cannot be used.

**Comment by TkDodo** [comment]
sorry, I'm not following. Hashing is only used internally. You'll always get the `queryKey` array "as is" passed into the queryFn.
